### PR TITLE
Update spacer.rb to allow building from source

### PR DESCRIPTION
**Pull request by tisba**
See https://github.com/samwho/spacer/issues/11

I'm not sure if this brew Formular was generated or not. If that's the case, let me know where and how and I'll take a look how to get these changes in there.

I think this is more like a bandaid and offering pre-compiled binaries at least for arm64-darwin would be great (compiling with Rust is slow and Apple Silicon will become dominant quickly).

What this PR does:

* **on intel/x86**: same behaviour as before (pre-build binaries will be downloaded from the release)
* **on arm**: as there are no pre-build binaries, configure the Formula to build from source using cargo.

In other words: On macOS on Apple Silicon (or Linux aarch) `brew install spacer` will always compile, while on x86 it will download the pre-build binaries.

In general the changes also allow to build from source regardless of architecture, if you want, and in addition build from head too:

```terminal
brew install --HEAD spacer
brew install --build-from-source spacer
```

## Also

* the indentation was of, which I fixed
* use double quotes consistently for String literals

### CODE REVIEW

**Review by samwho — APPROVED** [review]
This is awesome, thank you so much.

The formula was not generated, it was just the first time I'd ever written a formula so I copied one from another Rust tool. I forget which.

I'm shying away from doing ARM builds because the only way I know of doing them at the moment in GitHub Actions is through QEMU, which is horrendously slow. I'm open to suggestion.
